pkg/base1/cockpit.js

```diff
@@ -1047,6 +1047,26 @@ function basic_scope(cockpit, jquery) {
      *    immediately. Wait until control is returned to the browser.
      */
 
+    var call_queue = [];
+    var call_timeout = null;
+
+    function call_drain() {
+        call_timeout = null;
+        var func;
+        for (;;) {
```

Comment:
> Also, adding (for whatever reason) something null/undefined to the array would result in almost impossible to debug behavior.

None of the callers do this. The callers are below. But since you brought it up, I've made the code more verbose to make it clear that this is not a general purpose facility.

> This will work if the function calls (and clears) exactly as many items on the stack as were on it when it started, i.e.: get length of call_queue first, then loop over those, call if not null, remove them all.

Good point. Fixed by replacing the stack.